### PR TITLE
Add integration with SonarCloud (SonarQube) and fix Codecov

### PR DESCRIPTION
**Pull request by meltsufin**

### CODE REVIEW

**Comment by artembilan** [comment]
Yeah... Something is still wrong, @meltsufin : 
```
The command "./mvnw -T 1.5C test -B -P codecov sonar:sonar ${INTEGRATION_TEST_FLAGS}" exited with **1.**
```

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +14,13 @@ cache:
...
     - $HOME/google-cloud-sdk
     - $HOME/.m2
+addons:
+  sonarcloud:
+    organization: "spring-cloud-gcp"
+    token:
+      secure: "eTxYYXrExigaf4ELiNf87NjK//bu49tss5k8+l1+kKgyhsfGTP5aOE24ICJ9K7waBK5MV+P520s7oPhhMPjLJDxququDyTxoPV5xO9TAwVSqSSZwYfwRCPsGvlylP9NE/MCc211l6CihFrpJDb5pfJjndG64n3MBG691snTY5HXoWoZ0HarIAJc4KwhovFwqImL2+u5HOXogb+0y/kJsI/6qV+K16rC5Z5tNY3gIR9mrHvJvRgFzeZsZffCxJD1tJjNVsLoFBcTa2blpDlXd66dfCghGUlU30h7zm4WxD5J3R4hhrOvDzXdk4FFt2TPFz3TNbSf/MOEs/AMVSZH1SMH9ZXiBeBS86dGGWv+dFO5cxbP9Cg/lcL5JPupGYh0KSuZ4B2j6nbr65x9owD2+bBhi6qcP1Es01YVEa3WqsQI+/dbvOi6IChBDrrc9J6CL1RVBTzQVKeVlIEDkx7Mw/ivkL6DKKOeFRklz6bbYNAw4Z5gDkPZ3LezYW1gfpbUs5sUAcm/QYLDAFN37GhYMedCqeMbbxyMm9aTWob2qWShOCuytvHbseLP7r3VrvPfoUucDNF6T9oNLdSQ3OpyUl12LD7TgPikOqd3OqoRmb1/p/mCNmf8vX5USAp4L7G9edK4TzjZSwBcbRGZGqLrNvinwtCECP4wHJWPIzV33K90="
 script:
-  - ./mvnw -T 1.5C test -B -P codecov ${INTEGRATION_TEST_FLAGS}
+  - ./mvnw -T 1.5C test -B -P codecov sonar:sonar ${INTEGRATION_TEST_FLAGS}
```
[the remark this answers is not in the document]
Comment (meltsufin) [reply .travis.yml]: AFAIK, not necessarily, because Maven can resolve plugin goal executions based on FQN.

**Comment by artembilan** [comment]
OK. Now it's something different:
```
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.5.0.1254:sonar (default-cli) on project spring-cloud-gcp: Project was never analyzed. A regular analysis is required before analyzing a pull request 
```

**Comment by artembilan** [comment]
Don't we also need to add a ` - '$HOME/.sonar/cache'` into the `cache: directories:`?

See here: https://github.com/SonarSource/sq-com_example_java-maven-travis/blob/master/.travis.yml

**Comment by meltsufin** [comment]
@artembilan Looks like we're green now. :smile: 
Can you please turn on the [SonarCloud app](https://github.com/apps/sonarcloud)?

**Pull request merged**